File: native with local db and server/server/backend/events/repository/user_repository.py
from typing import Optional

from ..models import ApplicationUser


class UserRepository:

    def get_user(self, user_name: str) -> Optional[ApplicationUser]:
        try:
            return ApplicationUser.objects.get(pk=user_name)
        except Exception:
            return None

    def add_user(self, user_name: str, password: str, full_name: str) -> bool:
        user = self.get_user(user_name)
        if user is not None:
            return False
        user = ApplicationUser(username=user_name,
                               password=password,
                               full_name=full_name)
        user.save()
        return True

File: native with local db and server/native/app/src/main/java/com/example/cityinyourpocket/data_base/repository/UserRepository.kt
package com.example.cityinyourpocket.data_base.repository

import com.example.cityinyourpocket.data_base.dao.UserDao
import com.example.cityinyourpocket.data_base.entities.UserEntity

class UserRepository (private val userDao: UserDao){

    suspend fun createUser(userName: String, fullName: String, password: String){
        val entity = UserEntity(userName, fullName, password)
        userDao.addUser(entity);
        println(entity);
    }

    suspend fun verifyLogIng(userName: String, password: String): Boolean{
        val user = userDao.getUserByCredentials(userName, password) ?: return false
        return true;
    }

    suspend fun getUserByUserName(usrName: String): UserEntity?{
        return userDao.getUser(usrName)
    }
}